nengo_gui/components/component.py

```diff
@@ -1,49 +1,111 @@
 import json
 
 class Component(object):
-    def __init__(self, page, config, uid, component_order=0):
-        self.config = config
-        self.uid = uid
-        self.page = page
-        # the order this component will be defined in the javascript
+    """Abstract handler for a particular Component of the user interface.
+
+    Each part of the user interface has part of the code on the server-side
+    (in Python) and a part on the client-side (in Javascript).  These two sides
+    communicate via WebSockets, and the server-side is always a subclass of
+    Component.
+
+    Each Component can be configured via the nengo.Config system.  Components
+    can add required nengo objects into the model to allow them to gather
+    the required data from the running model.  Communication from server to 
+    client is done via Component.update_client(), which is called regularly
+    by the Server.ws_viz_component handler.  Communication from client to
+    server is via Component.message().
+    """
+
+    # The parameters that will be stored in the .cfg file for this Component
+    # Subclasses should override this as needed.
+    config_params = dict(x=0, y=0, width=100, height=100, label_visible=True)
+
+    def __init__(self, component_order=0):
+        # when generating Javascript for all the Components in a Page, they
+        # will be sorted by component_order.  This way some Components can
+        # be defined before others.
         self.component_order = component_order
+
+        # If we have reloaded the model (while typing in the editor), we need
+        # to swap out the old Component with the new one.
         self.replace_with = None
 
+    def initialize(self, page, config, uid):
```

Comment:
> initialize as a method name always make me cringe, since it seems like it should be done in __init__.

:)  fair enough.

I could see ```connect``` working, although that might be a bit ambiguous with the actual socket connections and nengo connections going on.  Perhaps ```attach```?  or ```set_page```?